### PR TITLE
[no sq] Misc fixes

### PR DESCRIPTION
**Pull request by sfan5**
commit 1:
even though it probably doesn't even work correctly these days, this would prevent anyone from creating a new game called "mesetint".

commit 2:
this was added in ee698770b98145bbb39f8671f8586e835b2e9ab4 and promptly never document.
there are other more standard env vars we respect and this is one less thing we have to rename.

commit 3:
_asdf_

commit 4:
VLAs are bad

## To do

This PR is Ready for Review.

### CODE REVIEW

**Comment by swagtoy** [comment]
Speaking of the Mesetint thing, should the hardcoded `devtest` warning be removed too? I think it's just a few lines sitting somewhere

File: src/gui/modalMenu.cpp [hunk 1]
```diff
@@ -306,13 +306,13 @@ bool GUIModalMenu::preprocessEvent(const SEvent &event)
 
 			s32 selected_idx = dropdown->getSelected();
 			s32 option_size = dropdown->getItemCount();
-			std::string list_of_options[option_size];
+			std::vector<std::string> list_of_options;
 
 			for (s32 i = 0; i < option_size; i++) {
-				list_of_options[i] = wide_to_utf8(dropdown->getItem(i));
+				list_of_options.push_back(wide_to_utf8(dropdown->getItem(i)));
```
Comment (nerzhul) [inline src/gui/modalMenu.cpp:312]: it's mostly the same but does more allocations, why change this ?

File: src/porting_android.cpp [hunk 2]
```diff
@@ -157,7 +157,7 @@ void showTextInputDialog(const std::string &hint, const std::string &current, in
 			jhint, jcurrent, jeditType);
 }
 
-void showComboBoxDialog(const std::string optionList[], s32 listSize, s32 selectedIdx)
+void showComboBoxDialog(const std::string *optionList, s32 listSize, s32 selectedIdx)
```
Comment (nerzhul) [inline src/porting_android.cpp:160]: warning we can now pass null there

File: src/gui/modalMenu.cpp (hunk 1 above)
Comment (sfan5) [reply src/gui/modalMenu.cpp:312]: like the commit says: to avoid VLAs

File: src/porting_android.cpp (hunk 2 above)
Comment (sfan5) [reply src/porting_android.cpp:160]: As far as I know `T[]` and `T*` are actually the exact same in function arguments, and both nullable.

**Comment by sfan5** [comment]
> Speaking of the Mesetint thing, should the hardcoded `devtest` warning be removed too?

Considering devtest exists as a game right now and the warning still serves its purpose: I disagree.

File: src/main.cpp [hunk 3]
```diff
@@ -938,14 +945,14 @@ static bool auto_select_world(GameParams *game_params)
...
-		dstream <<_("Automatically selecting world at") << " ["
+		dstream << "Automatically selecting world at ["
 		        << world_path << "]" << std::endl;
 	// If there are multiple worlds, list them
 	} else if (worldspecs.size() > 1 && game_params->is_dedicated_server) {
-		std::cerr << _("Multiple worlds are available.") << std::endl;
-		std::cerr << _("Please select one using --worldname <name>"
-				" or --world <path>") << std::endl;
-		print_worldspecs(worldspecs, std::cerr);
+		rawstream << "Multiple worlds are available.\n"
```
Comment (sfence) [inline src/main.cpp:952]: Why `std::endl` in one case and `\n` in another?

File: src/gui/modalMenu.cpp (hunk 1 above)
Comment (appgurueu) [reply src/gui/modalMenu.cpp:312]: Getting rid of these allocations is 100% premature optimization.

**Review by sfence — APPROVED** [review]

File: src/main.cpp (hunk 3 above)
Comment (sfan5) [reply src/main.cpp:952]: `endl` flushes the stream. No need to do that multiple times if we're printing a bunch of text.
(though I think with our current logging infrastructure there is no difference)

File: src/porting_android.cpp (hunk 2 above)
Comment (appgurueu) [reply src/porting_android.cpp:160]: sfan5 is right: https://stackoverflow.com/questions/10760893/c-vs-as-a-function-parameter

**Pull request merged**